src/supervisor/message_handlers.py

```diff
@@ -163,6 +171,239 @@ def _build_gzip_attachment(self, input_path):  #pylint: disable=no-self-use
+        mail.attachment = attachments
+        response = self.sendgrid_client.client.mail.send.post(request_body=mail.get())  # pylint: disable=unused-variable
+
+        #: Maybe test the response via response.status_code?
+
+    def _verify_addresses(self):
+        """Make sure from/to address keys exist and are not empty
+
+        Returns:
+            tuple (str): from and to addresses
+        """
+
+        try:
+            from_address = self.sendgrid_settings['from_address']
+            to_addresses = self.sendgrid_settings['to_addresses']
+
+        except KeyError:
+            warnings.warn('To/From address settings do not exist. No emails sent.')
+            return None, None
+
+        if not from_address or not to_addresses:
+            warnings.warn('To/From address settings exist but are empty. No emails sent.')
+            return None, None
+
+        return from_address, to_addresses
+
+    def _build_recipient_addresses(self, to_addresses):  #pylint: disable=no-self-use
+        """Craft 'to' addresses into a list of SendGrid 'To' objects
+
+        Args:
+            to_addresses (str or List): 'to' addresses, either as a single string or list of strings
+
+        Returns:
+            list (To): 'To' objects for future Mail() object
+        """
+
+        #: If we just get a string just return that one
+        if isinstance(to_addresses, str):
+            return [To(to_addresses)]
+
+        recipient_addresses = []
+        for address in to_addresses:
+            recipient_addresses.append(To(address))
+        return recipient_addresses
+
+    def _build_subject(self, message_details):
+        """Add prefix to subject if needed
+
+        Args:
+            message_details (MessageDetails): Will use .prefix if present
+
+        Returns:
+            str: Subject for outgoing email
+        """
+
+        subject = message_details.subject
+        if 'prefix' in self.sendgrid_settings:
+            subject = self.sendgrid_settings['prefix'] + subject
+
+        return subject
+
+    def _build_content(self, message, project_name):  #pylint: disable=no-self-use
+        """Add client version if desired and package into plaintext Content object
+
+        Args:
+            message (str): Main message content
+            project_name (str): pip-install name of the client project for client version number reporting
+
+        Returns:
+            Content: Content of email as a SendGrid Content object
+        """
+
+        #: Get the client's version (assuming client has been installed via pip install and setup.py)
+        distributions = pkg_resources.require(project_name)
+        if distributions:
+            version = distributions[0].version
+            version = f'\n\n{project_name} version: {version}'
+            message += version
+
+        return Content('text/plain', message)
+
+    def _verify_attachments(self, attachments):  #pylint: disable=no-self-use
```

Comment:
There are 4 method types to use here. Instance, class, static, or standalone. The instance method type doesn't really make sense because it's not accessing class instance state. Class methods don't really make sense because the method doesn't require class state. Static or stand alone would be the best fit.

Reply:
Yeah, I realize these are a little out of place as class methods. My thinking was that while they don't access/share state, they are specific to their associated MessageHandler and thus keeping them class methods makes sense from a organizational/readability standpoint. Does that have negative impacts/implications down the road?

Reply:
Making it static would keep the organization, remove the unused self, and make it explicitly clear that this method does not modify state. But no, I can't think of any negative impacts - Maybe some future refactor might end up modifying state when you don't expect it to?

Reply:
Hmm, I didn't fully appreciate `@staticmethod`s before. That definitely seems like the appropriate angle to go hear. Thanks for the heads up!